adam/learner/__init__.py

```diff
@@ -322,10 +324,26 @@ def concepts_to_patterns(self) -> Dict[Concept, PerceptionGraphPattern]:
 
     @abstractmethod
     def propose_updated_hypotheses(
-        self, concept_to_updated_patterns: Dict[Concept, PerceptionGraphPattern]
+        self,
+        concept_to_updated_patterns: Dict[
+            Concept, Dict[PerceptionGraphTemplate, PerceptionGraphPattern]
+        ],
     ) -> None:
         """
         Propose new/updated hypotheses to the learner.
 
```

Comment:
A bit more documentation for good measure

```suggestion

    This expects hypothesis updates to be given as mappings of the form `old_hypothesis -> new_hypothesis` where `old_hypothesis` is a `PerceptionGraphTemplate` and `new_hypothesis` is a `PerceptionGraphPattern`.

```